File: WorkSessons/WorkSesson8Lesson2.lua
App_Balance = 32
AmeliaMoney = 40
UserName = "Amelia"

print("Welcome to Cafe Curtis App")

function EnterCard()
    print("Enter your 6-digit gift card number:")
    Card_Code = io.read()
    if Card_Code == "123456" then
        MoneyDeposit()
    else
        print("Error wrong card number. Try again")
        EnterCard()
    end
end

function MoneyDeposit()
    print("Hello "..UserName)
    print("How much money would you like to add to your Cafe Curtis App?")
    MoneyAmount = io.read()
    MoneyAmount = tonumber(MoneyAmount)
    if MoneyAmount > AmeliaMoney then
        print("Error you don't have enough money to add that to your Cafe Curtis App BROKIEEEE try again broke ass! bum!!")
        MoneyDeposit()
    else
        App_Balance = App_Balance + MoneyAmount
        AmeliaMoney = AmeliaMoney - MoneyAmount
        print("Your Cafe Curtis App balance is now "..App_Balance.." The amount of money in your bank account is "..AmeliaMoney)
    end
end

EnterCard()

File: WorkSessons/WorkSesson8Lesson2.py
Card_Balance = 10
AmeliaWallet = 20

print("Welcome to the Cafe Curtis App!")

def EnterCode():
    CardCode = input("Enter your 6 digit gift card number")
    if CardCode == "123456":
        print("Access granted \n")
        MainThing()
    else:
        print("This did not work try again")
        EnterCode()





def MainThing():
    global Card_Balance, AmeliaWallet
    MoneyAmount = input("How much money would you like to add to your Cafe Curtis GiftCard?")
    MoneyAmount = int(MoneyAmount)
    if MoneyAmount > AmeliaWallet:
        print("Error you don't have that much money to put into your gift card.. HAHAHA BROKE LOSER!.. try again")
        MainThing()
    else:
        AmeliaWallet -= MoneyAmount
        Card_Balance += MoneyAmount
        print(f"Success you have {Card_Balance} amount of funds in your gift card and {AmeliaWallet} in your wallet")

EnterCode()